### PR TITLE
[DO NOT SUBMIT] Oneof -> inheritance take 2

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
This is a redesign of the oneof -> inheritance conversion, PTAL

File: test/testdata/java_book_names.baseline [hunk 1]
```diff
@@ -14,17 +14,20 @@
 
 package com.google.example.library.v1;
 
-import com.google.api.resourcenames.ResourceNameType;
+import com.google.api.resourcenames.ResourceName;
 
 // AUTO-GENERATED DOCUMENTATION AND CLASS
 @javax.annotation.Generated("by GAPIC protoc plugin")
-public class BookNameType implements ResourceNameType {
+public abstract class BookNames {
```
Comment (michaelbausor) [inline test/testdata/java_book_names.baseline]: Why is this abstract?

Comment (garrettjonesgoogle) [reply test/testdata/java_book_names.baseline]: Do allow for the addition of new methods in the future without breaking implementations

Comment (michaelbausor) [reply test/testdata/java_book_names.baseline]: Implementations? Isn't this a container for static methods?

Comment (garrettjonesgoogle) [reply test/testdata/java_book_names.baseline]: Oh crap you're totally right, this shouldn't be abstract.

**Comment by garrettjonesgoogle** [comment]
I fixed the `abstract` problem, PTAL

**Comment by garrettjonesgoogle** [comment]
The 2.7 build is failing, but I think only for line length. I'm inclined to ignore this, given that this is likely the last or close to last change in this repo...

**Pull request merged**